### ISSUE TITLE
Auth not resolving user for every request

### ISSUE THREAD

**Issue by megasteve19** [laravel/framework#46308]
- Laravel Version: 10.0.3
- PHP Version: 8.1.16
- Database Driver & Version: Postgres & I don't know, using sail

### Description:
When creating http kernel multiple times within same environment produces that auth user will not resolved anymore. First I used default kernel to make sure passport user can authanticate, but even token is diffrent user not changed for every request.
I solved this problem by making a seperate command to make internal request. Note that this command runs as different process.
```php
$request = Request::create(route('landlord.authentication.tenant.account.show'));

$request->headers->add([
    'X-TENANT-ID' => $this->argument('tenant-id'),
    'Authorization' => sprintf('Bearer %s', $this->argument('token')),
    'Accept' => 'application/json',
]);

$response = app(Kernel::class)->handle($request);

if ($response->isSuccessful())
{
    return $this->line(json_encode(['user' => Arr::get(json_decode($response->getContent(), true), 'data.id')]));
}

$this->line(json_encode(['user' => null]));
```

Below code is the driver I'm using for the socket server. This driver works fine for first connection. But on the second connection it will not work at all, it will resolve first connections user all the time. Until restarting the server. First I tought maybe the problem is customized kernel and request for websocket but I don't think so. If it was because of customization, default kernel should worked fine.
```php
// Using OpenSwoole.
Auth::viaRequest(
    'socket',
    fn (Request $request) => User::find($manager->findUserIdFromSocketId($request->fd()))
);
```

To sum it up, dispatching multiple requests within same environment/process does not resolves auth user for every request.

### Steps To Reproduce:
Create fresh Laravel project. Add few users. Add test command, for each users create kernel and dispatch request. Return authenticated user in the response. You will see that first will work as expected. But on the other responses you will see that it will return same user again.

Note: I tried `Facade::clearResolvedInstances();`. In a few hours I will create repo with reproduced issue.

**Comment by crynobone** [comment]
That should be expected. You can review `Illuminate\Http\Request::createFrom()` as example

**Comment by megasteve19** [comment]
@crynobone Thanks, I guess still having the same issue but I will look what I'm doing wrong.
